### DISCUSSION TITLE
Using web3wallet, how to set current chain when pairing with dapp

### DISCUSSION THREAD

**Discussion by mihoward21** [WalletConnect/walletconnect-monorepo#5276] [Developer Support]
In my wallet, I'm using the `buildApprovedNamespaces` util function to construct the namespaces when pairing with a dapp. I can properly set all my supported chains. But when connecting to the dapp, I want to tell the dapp what chain the user is currently set to in the wallet.

Could not find anything in the docs about this. Is there a recommended way to do this? Should we emit a chain changed event after the connection is successful? Something else?

**Comment by ronaldoguedess** [comment]
I guess that one Wallet can't change the dApp's chain directly. WalletConnect only establishes a connection across multiple addresses and chains, but it doesn't automatically update or reflect those changes within the dApp itself.

**Comment by mihoward21** [comment]
For anyone else looking at this, WalletConnect actually recommends not having any chain selection in the wallet. You include your supported chains when building the namespace, and that's all. Trust Wallet is an example of how this is done.
